src/app/carbon-estimator-form/carbon-estimator-form.component.html

```diff
@@ -1,4 +1,4 @@
-<form (ngSubmit)="handleSubmit()" [formGroup]="estimatorForm" class="tce-w-full tce-flex tce-flex-col tce-gap-6">
```

Comment:
I couldn't find a nice way to make the "submit" type "Calculate" button trigger conditionally based on whether the form was valid so I changed it to type="button" and set handleSubmit as the click handler with some extra logic so it would only submit the form when it was valid.

Reply:
This means that you don't get the diagram updating when you hit enter on a text form, which I think would be nice to keep. Is there anything preventing us from keeping this as the submit handler and just have the same early out on whether the form is valid?

Reply:
Yes, we can do that. We may need to find a way to provide feedback in this scenario when the form is invalid and hence not submitted when pressing the enter key. Otherwise it could be confusing if a user is working on a valid portion of the form but hitting enter doesn't do anything because another part of the form is invalid.

Reply:
I've reverted to using ngSubmit to trigger the handler so pressing enter triggers submission provided the form is valid.

Reply:
> We may need to find a way to provide feedback in this scenario when the form is invalid and hence not submitted when pressing the enter key.
Maybe make another spike around that? I'm not sure what's recommended and it is a bit inconsistent when Enter is going to submit the form vs select an option from a dropdown with keyboard controls etc. Was surprised that Enter submits the form when you have a checkbox focussed.